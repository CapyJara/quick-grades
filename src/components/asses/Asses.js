import React from 'react';
import PropTypes from 'prop-types';
import Ass from './Ass';
import styles from './asses.css';

function Asses({ asses }) {
  return  (
    <div className={styles.assBoard}>
      <h2>Submitted but not yet graded</h2>
      <ul>
        {asses.map((ass, i) => {
          return (<Ass key={i} ass={ass} />);
        })}
      </ul>
    </div>
  );
}

Asses.propTypes = {
  asses: PropTypes.oneOfType([PropTypes.bool, PropTypes.array])
};

export default Asses;
