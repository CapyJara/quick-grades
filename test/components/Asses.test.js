import React from 'react';
import { shallow } from 'enzyme';
import Asses from '../../src/components/asses/Asses';

describe('App component', () => {
  it('renders App', () => {
    const asses = [
      {
        gradeUrl: 'http',
        assignmentName: 'todo'
      },
      {
        gradeUrl: 'http',
        assignmentName: 'todo'
      },
      {
        gradeUrl: 'http',
        assignmentName: 'todo'
      }
    ];
    const wrapper = shallow(<Asses asses={asses} />);
    expect(wrapper).toMatchSnapshot();
  });
});
