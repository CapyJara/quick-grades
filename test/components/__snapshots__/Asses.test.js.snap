// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<div
  className="assBoard"
>
  <h2>
    Submitted but not yet graded
  </h2>
  <ul>
    <Ass
      ass={
        Object {
          "assignmentName": "todo",
          "gradeUrl": "http",
        }
      }
      key="0"
    />
    <Ass
      ass={
        Object {
          "assignmentName": "todo",
          "gradeUrl": "http",
        }
      }
      key="1"
    />
    <Ass
      ass={
        Object {
          "assignmentName": "todo",
          "gradeUrl": "http",
        }
      }
      key="2"
    />
  </ul>
</div>
`;
